core-common/src/main/java/org/apache/kylin/common/util/CliCommandExecutor.java

```diff
@@ -163,4 +163,34 @@ private void copyRemote(String localFile, String destDir) throws IOException {
         }
     }
 
+    public static final String COMMAND_INJECT_REX = "[ &`>|{}()$;\\-#~!+*”\\\\]+";
+
+    /**
+     * <pre>
+     * Check parameter for preventing command injection, replace illegal character into empty character.
+     *
+     * Note:
+     * 1. Whitespace is also refused because parameter is a single word, should not contains it
+     * 2. Some character may be illegal but still be accepted because commandParameter maybe a URI/path expression,
+     *     you may check "Character part" in https://docs.oracle.com/javase/8/docs/api/java/net/URI.html,
+     *     here is the character which is not banned.
+     *
+     *     1. dot .
+     *     2. slash /
+     *     3. colon :
+     *     4. equal =
+     *     5. ?
+     *     6. @
+     *     7. bracket []
+     *     8. comma ,
+     *     9. %
+     * </pre>
+     */
+    public static String checkParameter(String commandParameter) {
+        String repaired = commandParameter.replaceAll(COMMAND_INJECT_REX, "");
+        if (repaired.length() != commandParameter.length()) {
+            logger.debug("Detected illegal character in command {}, replace it to {}.", commandParameter, repaired);
```

Comment:
I think "debug" level here is improper; This might be an attack, it's log level should be info or warn.

Reply:
OK. The reason of I choose the debug level is for debug purpose, maybe the rare case if I mistake convert right command into some broken command.